src/main/java/org/folio/innreach/domain/service/impl/MARCTransformationOptionsSettingsServiceImpl.java

```diff
@@ -0,0 +1,100 @@
+    var marcTransformOptSet = findMARCTransformOptSet(centralServerId);
+    var updated = mapper.toMARCTransformationOptSet(marcTransformOptSetDTO);
+    updateMARCTransformOptSet(marcTransformOptSet, updated);
+    updateModifiedFields(marcTransformOptSet, updated);
+
+    repository.save(marcTransformOptSet);
+
+    return mapper.toMARCTransformationOptSetDto(marcTransformOptSet);
+  }
+
+  private void updateMARCTransformOptSet(MARCTransformationOptionsSettings marcTransformOptSet, MARCTransformationOptionsSettings updated) {
+    marcTransformOptSet.setExcludedMARCFields(updated.getExcludedMARCFields());
+    marcTransformOptSet.setConfigIsActive(updated.getConfigIsActive());
+  }
+
+  private void updateModifiedFields(MARCTransformationOptionsSettings marcTransformOptSet, MARCTransformationOptionsSettings updated) {
+    var currentModifiedFields = new ArrayList<>(marcTransformOptSet.getModifiedFieldsForContributedRecords());
+    var updatedModifiedFields = updated.getModifiedFieldsForContributedRecords();
+
+    // 1. Remove the existing database records that are no longer found in the incoming collection.
+    var modifiedFieldsToDelete = new ArrayList<>(currentModifiedFields);
+    modifiedFieldsToDelete.removeAll(updatedModifiedFields);
+    modifiedFieldsToDelete.forEach(marcTransformOptSet::removeModifiedFieldForContributedRecords);
+
+    // 2. Add the records found in the incoming collection, which cannot be found in the current database snapshot.
+    var newModifiedFields = new ArrayList<>(updatedModifiedFields);
+    newModifiedFields.removeAll(currentModifiedFields);
+    newModifiedFields.forEach(marcTransformOptSet::addModifiedFieldForContributedRecords);
+
+    // 3. Update the existing database records which can be found in the incoming collection.
+    updatedModifiedFields.removeAll(newModifiedFields);
+    for (FieldConfiguration updatedFieldConfiguration : updatedModifiedFields) {
+      updatedFieldConfiguration.setMARCTransformationOptionsSettings(marcTransformOptSet);
+      marcTransformOptSet.getModifiedFieldsForContributedRecords()
+        .set(marcTransformOptSet.getModifiedFieldsForContributedRecords().indexOf(updatedFieldConfiguration), updatedFieldConfiguration);
+    }
+  }
+
+  private MARCTransformationOptionsSettings findMARCTransformOptSet(UUID centralServerId) {
+    return repository.findOne(exampleWithServerId(centralServerId))
+      .orElseThrow(() -> new EntityNotFoundException("MARC Transformation Options Settings not found: " +
+        "centralServerId = " + centralServerId));
+  }
+
+  private static Example<MARCTransformationOptionsSettings> exampleWithServerId(UUID centralServerId) {
+    var toFind = new MARCTransformationOptionsSettings();
+    toFind.setCentralServer(centralServerRef(centralServerId));
+
+    return Example.of(toFind);
+  }
```

Comment:
` repository.findOne` with Example here produces an sql query with inner join of `central_server` which looks redundant as we don't need to map CentralServer entity to settings DTO anyway. And since there is cascade deletion on fk of central server id, we don't need an intersection of these entities to make sure that the server exists. 

I suggest adding custom repository method with query/named query to get better control on how entity is being loaded.
Something like this
```
  @QueryHints({ @QueryHint(name = org.hibernate.annotations.QueryHints.PASS_DISTINCT_THROUGH, value = "false") })
  @Query(value = "SELECT DISTINCT m FROM MARCTransformationOptionsSettings AS m where m.centralServer.id = :id")
  Optional<MARCTransformationOptionsSettings> findOneByCentralServerId(UUID id);
```